### ISSUE TITLE
Cannot use gmaps behind proxy

### ISSUE THREAD

**Issue by CNukem** [pbugnion/gmaps#250]
The official google maps services api solves this by allowing users to pass kwargs to requests, including proxy auths.

Related issue from googlemaps:
https://github.com/googlemaps/google-maps-services-python/issues/83

**Comment by pbugnion** [comment]
Thanks for raising this.

Does this apply to querying the directions service, or even just for rendering a map?

So I understand the use case correctly, there is an authenticating proxy that prevents *outbound* connections to the Google APIs from your browser?

**Comment by CNukem** [comment]
Definitely querying for directions, and I think just rendering as well.

Should 
```py
fig = gmaps.figure()
fig
```
Display anything if I've not added any layers?


All outbound requests must go through the proxy in this environment.

**Comment by evadeflow** [comment]
Just wanted to chime in here that gmaps is working fine for me behind a corporate HTTP proxy. The only tricky bit was that I had to do _both_ of the following:

1. Set `http_proxy` and `https_proxy` appropriately in the terminal in which I ran `jupyter notebook`. (i.e., ran: `export http_proxy='http://<username>:<password>@<proxy-ip>:<proxy-port/'; export https_proxy="${http_proxy}"`)
1. Configure my browser (Firefox) to use the same proxy.

Both were necessary. When I initially forgot to set the proxy vars, I was getting errors like:

```
usr/lib/python3.5/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1254                 h.request(req.get_method(), req.selector, req.data, headers)
   1255             except OSError as err: # timeout error
-> 1256                 raise URLError(err)
   1257             r = h.getresponse()
   1258         except:

URLError: <urlopen error [Errno -2] Name or service not known>
```

Once I set `http[s]_proxy`, I thought, "Oh, okay, makes sense. I'm _good_ now."  But then I was seeing a blank map in my notebook, which motivated me to comment over on issue #246. Configuring the browser to also use the same proxy was what finally made everything work...
